go/felix/iptables/table.go

```diff
@@ -366,23 +345,29 @@ func (t *Table) loadDataplaneState() {
 
 	// Now scan for chains that shouldn't be there and mark for deletion.
 	t.logCxt.Info("Scanning for unexpected iptables chains")
-	for chainName := range dataplaneHashes {
+	for chainName, dataplaneHashes := range dataplaneHashes {
 		logCxt := t.logCxt.WithField("chainName", chainName)
 		if t.dirtyChains.Contains(chainName) || t.dirtyInserts.Contains(chainName) {
 			// Already an update pending for this chain.
 			logCxt.Debug("Skipping known-dirty chain")
 			continue
 		}
-		if !t.ourChainsRegexp.MatchString(chainName) {
-			// Skip non-felix chain
-			logCxt.Debug("Skipping non-calico chain")
-			continue
-		}
 		if _, ok := t.chainToDataplaneHashes[chainName]; ok {
 			// Chain expected, we'll have checked its contents above.
 			logCxt.Debug("Skipping expected chain")
 			continue
 		}
+		if !t.ourChainsRegexp.MatchString(chainName) {
+			// Non-calico chain, check it doesn't have any rule insertions in it.
+			for _, hash := range dataplaneHashes {
+				if hash != "" {
+					logCxt.Info("Found unexpected insert, marking for cleanup")
```

Comment:
I'm not sure I understand why Calico inserts are unexpected here.  Is it because this method only runs at start of day?